rust/src/mqtt/mqtt_property.rs

```diff
@@ -0,0 +1,268 @@
+
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// written by Sascha Steinbiss <sascha@steinbiss.name>
+
+use crate::mqtt::parser::*;
+use crate::jsonbuilder::{JsonBuilder, JsonError};
+use nom::number::streaming::*;
+use nom::*;
+
+#[derive(Debug)]
+#[allow(non_camel_case_types)]
+pub enum MQTTProperty {
+    UNKNOWN,
+    PAYLOAD_FORMAT_INDICATOR(u8),
```

Comment:
Would it make sense to add detection on a kind of property present ?
Something like `mqtt.property: AUTHENTICATION_METHOD`

Reply:
Maybe in a later version, as a feature addition? Whether this is a must-have feature is a question that someone with more experience with the MQTT protocol in real-life practice should answer :)

Reply:
Could you simply add a comment about it for now ?

Reply:
Yes, of course.